### PR TITLE
Assorted (documentation) fixes for `POST /containers/{id}/wait`

### PR DESCRIPTION
**Pull request by thaJeztah**
(See individual commits for details)

### api/swagger: update /containers/{id}/wait "condition" parameter to match code

This patch updates the swagger, and:

- adds an enum definition to document valid values (instead of describing them)
- updates the description to mention both "omitted" and "empty" values (although the former is already implicitly covered by the field being "optional" and having a default value).

### api/swagger: add missing 400 response for POST /containers/{id}/wait

The `/containers/{id}/wait` can return a 400 (invalid argument) error if `httputils.ParseForm()` fails.

### client.ContainerWait(): don't send empty "condition" query parameter

The client would always send a value, even if no `condition` was set;

    Calling POST /v1.41/containers/foo/wait?condition=

This patch changes the client to not send the parameter if it's empty (and the API default value should be used).

### api: POST /containers/{id}/wait: validate "condition" parameter

The endpoint was silently ignoring invalid values for the "condition" parameter. This patch now returns a 400 status if an unknown, non-empty "condition" is passed.

With this patch:

```bash
curl --unix-socket /var/run/docker.sock -XPOST 'http://localhost/v1.41/containers/foo/wait?condition=foobar'
{"message":"invalid condition: \"foobar\""}
```

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by thaJeztah** [comment]
(added cherry-pick label for the documentation side of this)

**Comment by tianon** [comment]
Looks fine to me, but b888dcc81934339a1dce7b90c5698058741e28fe and 6126cdc4cfd55b4c4b4b96ce8b92805d579e298a both claim to be removing `required: true` (and neither appear to). :see_no_evil:

**Comment by thaJeztah** [comment]
> Looks fine to me, but https://github.com/moby/moby/commit/b888dcc81934339a1dce7b90c5698058741e28fe and https://github.com/moby/moby/commit/6126cdc4cfd55b4c4b4b96ce8b92805d579e298a both claim to be removing required: true (and neither appear to). 🙈

Good catch (silly mistake; I originally put the `enum` at the property above `condition`, and therefore thought it was marked as `required` - then I spotted my mistake, and moved the `enum` to the right location, but didn't update the description 😂)

Anyway; updated the commit messages 👍

**Review by ndeloof — APPROVED** [review]

**Review by tianon — APPROVED** [review]

**Comment by thaJeztah** [comment]
@cpuguy83 ptal

**Comment by cpuguy83** [comment]
So older versions of the client always set condition=?
Wouldn't this break that?

**Comment by cpuguy83** [comment]
Would be really nice if we didn't have two different types for the same condition.

**Comment by thaJeztah** [comment]
Current clients (CLI) unconditionally set `?condition=`, but the API already ignores it if it's empty.

The default is determined by the daemon (which is already the  case as well)

**Comment by cpuguy83** [comment]
And in that case, older clients will now get an error instead of being ignored?

**Comment by thaJeztah** [comment]
No, because we first check if the `condition` is set, and non-empty.

The only error _added_ is `?condition=SomeBogusValue`, which previously was silently ignored (and no client should send it currently, unless someone created their own and used an invalid value for `condition`)

File: api/server/router/container/container_routes.go [hunk 1]
```diff
@@ -336,12 +336,16 @@ func (s *containerRouter) postContainersWait(ctx context.Context, w http.Respons
 		if err := httputils.ParseForm(r); err != nil {
 			return err
 		}
-		switch container.WaitCondition(r.Form.Get("condition")) {
-		case container.WaitConditionNextExit:
-			waitCondition = containerpkg.WaitConditionNextExit
-		case container.WaitConditionRemoved:
-			waitCondition = containerpkg.WaitConditionRemoved
-			legacyRemovalWaitPre134 = versions.LessThan(version, "1.34")
+		if v := r.Form.Get("condition"); v != "" {
```
Comment (thaJeztah) [inline api/server/router/container/container_routes.go:339]: Here, we first check if it's non-empty before checking what value was passed

**Review by cpuguy83 — APPROVED** [review]
LGTM

File: api/server/router/container/container_routes.go [hunk 2]
```diff
@@ -336,12 +336,16 @@ func (s *containerRouter) postContainersWait(ctx context.Context, w http.Respons
...
+			switch container.WaitCondition(v) {
+			case container.WaitConditionNextExit:
+				waitCondition = containerpkg.WaitConditionNextExit
+			case container.WaitConditionRemoved:
+				waitCondition = containerpkg.WaitConditionRemoved
+				legacyRemovalWaitPre134 = versions.LessThan(version, "1.34")
+			default:
+				return errdefs.InvalidParameter(errors.Errorf("invalid condition: %q", v))
+			}
 		}
```
Comment (thaJeztah) [inline api/server/router/container/container_routes.go:349]: This block didn't account for situations where the default value was explicitly passed, causing a regression; fixing in https://github.com/moby/moby/pull/43747

**Pull request merged**